### PR TITLE
Move model capabilitiy verification out of openAIChat.m, for maintain…

### PR DESCRIPTION
**Pull request by ccreutzi**
Move model capability verification out of main openAIChat class, for maintainability

Also, fix that `bot=openAIChat(ResponseFormat="json")` did not set `bot.ResponseFormat`.

### CODE REVIEW

**Comment by ccreutzi** [comment]
> ## [Codecov](https://app.codecov.io/gh/matlab-deep-learning/llms-with-matlab/pull/37?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=matlab-deep-learning) Report
> Attention: Patch coverage is `17.41573%` with `147 lines` in your changes are missing coverage.

That's codecov missing coverage during test class creation.

**Review by vpapanasta — APPROVED** [review]
That's looking great, thanks for doing this Christopher.

File: openAIChat.m [hunk 1]
```diff
@@ -114,13 +114,7 @@
...
                 systemPrompt                       {llms.utils.mustBeTextOrEmpty} = []
                 nvp.Tools                    (1,:) {mustBeA(nvp.Tools, "openAIFunction")} = openAIFunction.empty
-                nvp.ModelName                (1,1) string {mustBeMember(nvp.ModelName,[...
-                                                            "gpt-4o","gpt-4o-2024-05-13",...
-                                                            "gpt-4-turbo","gpt-4-turbo-2024-04-09",...
-                                                            "gpt-4","gpt-4-0613", ...
-                                                            "gpt-3.5-turbo","gpt-3.5-turbo-0125", ...
-                                                            "gpt-3.5-turbo-1106",...
-                                                        ])} = "gpt-3.5-turbo"
+                nvp.ModelName                (1,1) string {mustBeModel} = "gpt-3.5-turbo"
```
Comment (debymf) [inline openAIChat.m:117]: Do we still get tab completion?

**Review by debymf — APPROVED** [review]

File: openAIChat.m (hunk 1 above)
Comment (ccreutzi) [reply openAIChat.m:117]: I was going to create a json file for tab completion. That should then also enable completion for other NVPs.

**Pull request merged**